src/World.cpp

```diff
@@ -1495,7 +1515,17 @@ bool cWorld::GrowRipePlant(int a_BlockX, int a_BlockY, int a_BlockZ, bool a_IsBy
 				{
 					return false;
 				}
-				FastSetBlock(a_BlockX, a_BlockY, a_BlockZ, BlockType, 7);
+
+				if (!a_IsByBonemeal)
+				{
+					++BlockMeta;
+				}
+				else
+				{
+					BlockMeta += random.NextInt(4) + 2;
+					BlockMeta = std::min(BlockMeta, (NIBBLETYPE)7);
```

Comment:
Ok. I wasn't sure if I needed to use the c++ cast templates for basic types. I know std::reinterpret_cast is used for pointer types, but I was unsure about basic types such as ints and unsigned chars.